src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,446 @@
+---
+title: PL/pgSQL
+summary: PL/pgSQL is a procedural language that you can use within user-defined functions and stored procedures.
+toc: true
+key: sql-expressions.html
+docs_area: reference.sql
+---
+
+{% include enterprise-feature.md %}
+
+{% include_cached new-in.html version="v23.2" %} [PL/pgSQL](https://www.postgresql.org/docs/16/plpgsql-overview.html) is a procedural language that you can use within [user-defined functions]({% link {{ page.version.version }}/user-defined-functions.md %}) and [stored procedures]({% link {{ page.version.version }}/stored-procedures.md %}) in CockroachDB. 
+
+In contrast to [SQL statements]({% link {{ page.version.version }}/sql-statements.md %}), which are issued one-by-one from the client to the database, PL/pgSQL statements are encapsulated in a [block structure](#structure) and executed on the database side, thus reducing network latency. PL/pgSQL enables more complex functionality than standard SQL, including [conditional statements](#write-conditional-statements), [loops](#write-loops), and [exception handling](#report-messages-and-handle-exceptions).
+
+This page describes PL/pgSQL [structure](#structure) and [syntax](#syntax), and includes [examples](#examples) of functions and procedures that use PL/pgSQL.
+
+## Structure
+
+A function or procedure that uses PL/pgSQL must specify the `PLpgSQL` language within the [`CREATE FUNCTION`]({% link {{ page.version.version }}/create-function.md %}) or [`CREATE PROCEDURE`]({% link {{ page.version.version }}/create-procedure.md %}) statement:
+
+~~~ sql
+CREATE [ PROCEDURE | FUNCTION ] ...
+  LANGUAGE PLpgSQL
+  ...
+~~~
+
+PL/pgSQL is block-structured. A block contains the following:
+
+- An optional `DECLARE` section that contains [variable declarations](#declare-a-variable) for all variables that are used within the block and are not defined as [`CREATE FUNCTION`]({% link {{ page.version.version }}/create-function.md %}) or [`CREATE PROCEDURE`]({% link {{ page.version.version }}/create-procedure.md %}) parameters.
+- A [function]({% link {{ page.version.version }}/user-defined-functions.md %}) or [procedure]({% link {{ page.version.version }}/stored-procedures.md %}) body, consisting of statements enclosed by `BEGIN` and `END`.
+- An optional `EXCEPTION` section for [catching and handling `SQLSTATE` errors](#write-exception-logic).
```

Comment:
Note: this goes before the `END`, after any other body statements.

Reply:
Yup, this is covered in the corresponding section.